hw1/task1/get_stats.py

```diff
@@ -0,0 +1,178 @@
+    invalid_rows += data['passenger_count'].isna().sum()
+
+    data = data.dropna(subset=['lpep_pickup_datetime',
+                               'lpep_dropoff_datetime',
+                               'passenger_count',
+                               'trip_distance',
+                               'total_amount'])
+
+    gen_stat['mean_cost'] = data['total_amount'].mean()
+    data['trip_duration'] = data['lpep_dropoff_datetime'] - data['lpep_pickup_datetime']
+    longest = data['trip_duration'].max()
+    gen_stat['longest_ride'] = longest
+
+    pickup_times = pd.DataFrame(data['lpep_pickup_datetime'])
+    if pickup_times.shape[0] != 0:
+        pickup_times.index = pickup_times['lpep_pickup_datetime']
+        max_count_start = pickup_times['lpep_pickup_datetime'].resample('10T').count().idxmax()
+        max_count = pickup_times['lpep_pickup_datetime'].resample('10T').count().max()
+        max_count_end = max_count_start + np.timedelta64(10, 'm')
+        gen_stat['max_count'] = max_count
+        gen_stat['max_count_start'] = max_count_start
+        gen_stat['max_count_end'] = max_count_end
+
+    gen_stat['invalid_rows'] = invalid_rows
+    gen_stat = pd.DataFrame(gen_stat, index=[0])
+    return gen_stat
+
+
+def missing_dates(path):
+
+    missing_date = pd.Series()
+    data, _ = read_value_data(path)
+    data.columns = np.vectorize(lambda s: s.lower())(data.columns)
+
+    data['lpep_pickup_datetime'] = data['lpep_pickup_datetime'].apply(to_datetime)
+    data = data.dropna(subset=['lpep_pickup_datetime'])
+
+    data['lpep_dropoff_datetime'] = data['lpep_dropoff_datetime'].apply(to_datetime)
+    missing_date = missing_date.append(
+        data['lpep_pickup_datetime'].iloc[np.where(data['lpep_dropoff_datetime'].isna())[0]].apply(lambda i: i.date()),
+        ignore_index=True)
+
+    data['trip_distance'] = data['trip_distance'].apply(to_float)
+    missing_date = missing_date.append(
+        data['lpep_pickup_datetime'].iloc[np.where(data['trip_distance'].isna())[0]].apply(lambda i: i.date()),
+        ignore_index=True)
+
+    data['total_amount'] = data['total_amount'].apply(to_float)
+    missing_date = missing_date.append(
+        data['lpep_pickup_datetime'].iloc[np.where(data['total_amount'].isna())[0]].apply(lambda i: i.date()),
+        ignore_index=True)
+
+    data['passenger_count'] = data['passenger_count'].apply(to_int)
+    missing_date = missing_date.append(
+        data['lpep_pickup_datetime'].iloc[np.where(data['passenger_count'].isna())[0]].apply(lambda i: i.date()),
+        ignore_index=True)
+
+    data = data.dropna(subset=['lpep_dropoff_datetime',
+                               'passenger_count',
+                               'trip_distance',
+                               'total_amount'])
+    valid_dates = data['lpep_pickup_datetime'].apply(lambda i: i.date())
+    missing_date = pd.DataFrame(missing_date.apply(lambda s: None if s in valid_dates.values else s).dropna(),
```

Comment:
it returns not correct result.
expected: all the dates without any valid record in [min,max] range, where min, max - valid min and max dates in valid records